### ISSUE TITLE
Consider warning for [StructLayout] usage with primary constructor

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#67162]
I noticed that we produce a warning if you use StructLayout on a partial (undefined ordering).
Should we do something similar for primary constructors (and record structs)?
Alternatively, we can define/specify/verify the ordering.

```
// (5,15): warning CS0282: There is no defined ordering between fields in multiple declarations of partial struct 'C'. To specify an ordering, all instance fields must be in the same declaration.                
Diagnostic(ErrorCode.WRN_SequentialOnPartialClass, "C").WithArguments("C")); 
```

This was also reported [here](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1748065) with an example showing an unexpected execution output.

Relates to test plan https://github.com/dotnet/roslyn/issues/65697

**Comment by jaredpar** [comment]
I agree that all three of these scenarios are related because they have the following properties:

1. The ordering of members in metadata are not specified by the language / compiler. 
2. The ordering is generally understood because the compiler is deterministic, generally it's the same as lexical order in the files, and we don't change the implementation between releases. 

If we were starting from scratch I'd probably push for an error here. Given that `partial` set the precedence though I think we should just follow on that here and issue a warning. Do think this warning should be tied to lang version 12 though.
